src/main.ts

```diff
@@ -80,19 +80,43 @@ const wait_for_input = () => {
   })
 }
 
+const timeout = (ms: number) => {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+const read_cache = async() => {
+  const cache = await Checkin.find()
+  await Promise.all(cache.map(async (checkin) => {
+    await send_check_in(checkin)
+  }))
+}
+
+const cache_observer = () => {
+  let is_resolved = true
+  setInterval(async()=> {
+    if (is_resolved) {
+      is_resolved = false
+      await read_cache()
+      is_resolved = true
+    }
+  }, (30 * 1000))
+}
+
+
 //Sends a checkin until the backend recieves it
-const send_check_in = async (checkin: Checkin, backoff: number) => {
-  await axios.post('/checkin', checkin)
-  .catch (async (error) => {
+const send_check_in = async (checkin: Checkin) => {
+  try{
+    await axios.post('/checkin', checkin)
+    await delete_check_in(checkin.student_id)
+  }
+  catch (error: any){
     const status = error.toJSON().status
     if (status >= 500 || status === null) {
-      await new Promise(f => setTimeout(f, backoff * 1000));
-      backoff *= 2
-      if (backoff > 128) { backoff = 128 }
-      send_check_in(checkin, backoff)
+      await send_check_in(checkin)
```

Comment:
I feel like we should maybe have some sort of falloff here... maybe add an `await timeout(10000)` so we don't spam the backend?

Do we even WANT to force retry? Should we just wait for the observer?